pycloudlib/azure/util.py

```diff
@@ -34,11 +35,19 @@ def get_client(resource, config_dict: dict):
     try:
         credential = AzureCliCredential()
         subscription_id = config_dict.get("subscriptionId")
-        return resource(credential, subscription_id=subscription_id)
+        client = resource(credential, subscription_id=subscription_id)
+        # smoke test credentials based on Azure CLI config if present
+        [_ for _ in client.operations.list()]
```

Comment:
Can we comment here that if this line fails we'll raise the ClientAuthenticationError so it doesn't look like a meaningless statement when we come back 3 months later?

Reply:
Also, do we actually need to build a list here or does calling `client.operations.list()` work?

Reply:
Yes unfortunately iterating on that list is what triggers. But we could limit that to just a single next()
```
(Pdb) client.operations.list()
<iterator object azure.core.paging.ItemPaged at 0x7fc76b9ff640>

(Pdb) _ = next(client.operations.list())
*** azure.identity._exceptions.CredentialUnavailableError: Azure CLI not found on path
```

alizing 
```suggestion
        _ = next(client.operations.list())
```